internal/client/blueprints.go

```diff
@@ -0,0 +1,167 @@
+// Package client - blueprints contains functions for the blueprint API
+// Copyright (C) 2020 by Red Hat, Inc.
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+// PostTOMLBlueprintV0 sends a TOML blueprint string to the API
+// and returns an APIResponse
+func PostTOMLBlueprintV0(socket, blueprint string) *APIResponse {
```

Comment:
Suggestion for future improvement:
It would be nice to pass in an already opened socket here, rather than a string, and also to pass in an Blueprint type and let the lower layers marshal it.

Reply:
I hadn't considered an open socket. I'm not sure how much benefit that would have, or if it even really matters one way or the other.

As for passing in the blueprint, that could be done at a higher layer. I was trying to keep this as generic as I could, only depending on the return types.

Reply:
In general, we try to open the sockets at the highest possible level, and not in the libraries. That way they can be more flexible (maybe for writing tests?). In practice it would probably not matter for this, would just be about consistency and typesafety.

As for the Blueprint type: if you use that rather than `string` and do the marshalling internally, then you enforce both that this sends actual blueprints, and that the marshalling is TOML, reflecting the name of the function.